pdns/backends/gsql/gsqlbackend.cc

```diff
@@ -2156,7 +2156,7 @@ void GSQLBackend::extractRecord(SSqlStatement::row_t& row, DNSResourceRecord& r)
 
   r.qtype=row[3];
 
-  if (d_upgradeContent && DNSRecordContent::isUnknownType(row[3]) && r.qtype.isSupportedType()) {
+  if (d_upgradeContent && DNSRecordContent::isUnknownType(row[3]) && r.qtype.isSupportedType() && !r.qtype.isMetadataType() && DNSRecordContent::isRegisteredType(r.qtype, r.qclass) && r.qtype != QType::NSEC && r.qtype != QType::NSEC3 && r.qtype != QType::ANY) {
```

Comment:
@mind04 pointed out to me that, while NSEC/NSEC3 should never be in the records table, if they are, we probably do still want to convert them.

Reply:
how about: `d_upgradeContent && DNSRecordContent::isUnknownType(row[3]) && r.qtype.isSupportedType() && !r.qtype.isMetadataType() && r.qclass == QClass::IN && r.qtype != QType::ANY`

Reply:
because I now realise we don't really care whether the type and class are valid together; our database is strictly class IN

Reply:
I missed that `registered` actually checks that we can convert it from/to text. So, perhaps it can be as short as: `d_upgradeContent && DNSRecordContent::isUnknownType(row[3]) && DNSRecordContent::isRegisteredType(QClass::r.qtype, r.qclass)`

Reply:
I was typing a comment and then forgot to post it. Looking at the code, `regist` skips the typemap registration because of the `nullptrs`, which indeed matches @mind04's observation.

Reply:
Alright, thanks! I just applied the change.

Reply:
I don't see it

Reply:
Ugh, I had pushed it to my own fork *again*, sorry! Sleep really is underrated.

Reply:
Please feel free to do this PR against your own fork if that makes your life easier, @rgacogne

Reply:
thanks Jacob, I believe we're ready to merge :)